src/signal_wrap.cc

```diff
@@ -90,6 +90,7 @@ class SignalWrap : public HandleWrap {
     SignalWrap* wrap;
     ASSIGN_OR_RETURN_UNWRAP(&wrap, args.Holder());
     int signum = args[0]->Int32Value();
+    bool oneshot = args[1]->BooleanValue();
```

Comment:
Can you `CHECK(args[1]->IsBoolean());` and use `args[1].As<Boolean>()->Value()` instead?

(Or use the overload of `BooleanValue()` that takes a `context` argument?)